.github/workflows/tf-lint.yaml

```
@@ -0,0 +1,33 @@
+  pull_request:
+    branches:
+      - main
+
+jobs:
+  terraform_checks:
+    name: Run Terraform Lint and Format Check
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```

actionlint (1.7.4)

the runner of "actions/checkout@v2" action is too old to run on GitHub Actions. update the action's version to fix this issue

(action)